cmd/soroban-cli/src/commands/contract/read.rs

```diff
@@ -112,20 +113,28 @@ impl Cmd {
             last_modified_ledger,
         } in &entries.entries
         {
-            let (
-                LedgerKey::ContractData(LedgerKeyContractData { key, .. }),
-                LedgerEntryData::ContractData(ContractDataEntry { val, .. }),
-            ) = &(key, val)
-            else {
-                return Err(Error::OnlyDataAllowed);
+            let (key, val) = match (key, val) {
```

Comment:
One problem with us allowing the `read` command to output the key and value of contract code, is that the outputs are now of different types. Before the outputs were always keys and values for contract data. Now it's also contract code.

When I opened #1318 I was confused because I was thinking of the read command reading raw ledger entries, but that's not what the read command does, it's a `contract read` command and it is only reading contract data, not all artbirary ledger entries.

There exists `contract fetch` for reading wasm.

It's confusing that the read command has a `--wasm-hash` parameter. I'm not sure why that it needs a parameter for restoring the contract.

I think my expectations in #1318 are incorrect, and the expectation should be probably to:
1. Remove the `--wasm-hash` and `--wasm` options after understanding why they are present.
1. Fix the error to say that only contract data can be read instead of data and code.